server/users.ts

```diff
@@ -526,6 +526,7 @@ class User extends Chat.MessageContext {
 		this.lastPM = '';
 		this.team = '';
 		this.lastMatch = '';
+		this.forcedPublic = null;
```

Comment:
This is actually another reason to use `null` over `undefined`; according to various docs on how V8 works, forcing it to be set here will make it so `User` stays the same virtual class, instead of V8 needing to change the class when the property is added.

Reply:
It matters what order you add properties, too, because it affects stuff like the order of `Object.keys`.

If you set `forcedPublic` first, vs if you set `locked` first, would make them end up as different classes (as verified by `Object.keys`), leading to significantly more polymorphism needed by functions taking `User` objects than you would expect.

Reply:
The class is already megamorphic

Reply:
In theory, I want to fix that, and TypeScript can assist in doing so.